### ISSUE TITLE
consider OSC 777 and OSC 9, OSC 99

### ISSUE THREAD

**Issue by davidosomething** [ObserverOfTime/notifications.nvim#1]
these are special escape sequences you output to the terminal
then your terminal emulator can handle the notification (either by itself, or forwarding to your system notifications)

wezterm, foot, rxvt support 777

kitty uses 99 https://sw.kovidgoyal.net/kitty/desktop-notifications/

**Comment by ObserverOfTime** [comment]
Looks like they don't support icons or urgencies so I don't know if it's worth implementing.

**Comment by kovidgoyal** [comment]
I can certainly add support for urgency level to OSC 99 however icons
are more problematic. How would you expect them to work? Paths wont work
over SSH for example.

**Comment by ObserverOfTime** [comment]
I suppose SSH is a valid use case (will still be missing urgencies though). I'll look into it ~~today~~ tomorrow.

**Comment by kovidgoyal** [comment]
And I added support for urgency to OSC 99: https://github.com/kovidgoyal/kitty/commit/865f66221667117ab4d555bfd4bb81b4856f2a18

**Comment by ObserverOfTime** [comment]
> And I added support for urgency to OSC 99: [kovidgoyal/kitty@865f662](https://github.com/kovidgoyal/kitty/commit/865f66221667117ab4d555bfd4bb81b4856f2a18)

Will it throw an error if the urgency is specified in an older version?

**Comment by kovidgoyal** [comment]
On Thu, May 16, 2024 at 09:15:49AM -0700, ObserverOfTime wrote:
> > And I added support for urgency to OSC 99: ***@***.***(https://github.com/kovidgoyal/kitty/commit/865f66221667117ab4d555bfd4bb81b4856f2a18)
>
> Will it throw an error if the urgency is specified in an older version?

No.

**Comment by ObserverOfTime** [comment]
Implemented in 4ae7b6278f1699e3e3500acc4824ea3907660d08. Must be enabled by setting `vim.g.notifications_use_osc = ...` (replace `...` with `'9'`, `'99'`, or `'777'`) before calling `notifications.setup`. I will not add it to the setup opts or attempt to autodetect the terminal.

**Comment by davidosomething** [comment]
why is `nvimcord` and not `notifications` ? also seems to be broken or missing some nvimcord dependency

**Comment by ObserverOfTime** [comment]
Typo. I was working on another plugin at the same time and my brain bugged out. Fixed in 3ca7f5e.
```lua
vim.g.notifications_use_osc = '777'
require('notifications').setup(...)
```

**Comment by davidosomething** [comment]
all good
